### PR TITLE
✨ Fix GetObjectVariableInto util func

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Review by lubronzhan — APPROVED** [review]

**Review by lubronzhan — APPROVED** [review]

File: exp/runtime/topologymutation/variables.go [hunk 1]
```diff
@@ -77,18 +76,13 @@ func GetObjectVariableInto(templateVariables map[string]apiextensionsv1.JSON, va
...
 		return errors.Wrapf(err, "failed to unmarshal variable json %q into %q", string(value.Raw), into)
 	}
 
 	return nil
 }
 
-func sanitizeJSON(input []byte) (output []byte) {
-	output = []byte(strings.ReplaceAll(string(input), "\\", ""))
-	return output
-}
```
Comment (vincepri) [inline exp/runtime/topologymutation/variables.go:90]: Why was this here?

File: exp/runtime/topologymutation/variables_test.go [hunk 2]
```diff
@@ -266,6 +267,22 @@ func Test_GetVariableObjectWithNestedType(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "valid variable with encoded character",
+			variables: map[string]apiextensionsv1.JSON{
+				// Note: When a user uses `<` in a string in a variable it will be encoded as `\u003c`
```
Comment (enxebre) [inline exp/runtime/topologymutation/variables_test.go:273]: Can you help me understand how is this relevant to dropping the sanitizeJSON func above?

Comment (sbueringer) [reply exp/runtime/topologymutation/variables_test.go:273]: The previous version of the func didn't work if `<` was part of the string because it was automatically encoded to `\u003c`.

In general just dropping `\\` is not a very good idea when unmarshalling JSON

Comment (sbueringer) [reply exp/runtime/topologymutation/variables_test.go:273]: Updated the comment slightly. Let me know if that makes sense so far and if I should explain more in the comment
